test/framework/modules.py

```diff
@@ -178,6 +178,18 @@ def test_load(self):
         for mod in mods:
             self.assertErrorRegex(EasyBuildError, '.*', self.modtool.load, [mod])
 
+        # by default, modules are always loaded, even if they are already loaded
+        self.modtool.load(['GCC/4.7.2'])
+        del os.environ['EBROOTGCC']
+        self.assertTrue('GCC/4.7.2' in self.modtool.loaded_modules())
+        self.modtool.load(['GCC/4.7.2'])
+        self.assertTrue(os.environ.get('EBROOTGCC'))
+
+        # reloading can be disabled using allow_reload=False
+        del os.environ['EBROOTGCC']
+        self.modtool.load(['GCC/4.7.2'], allow_reload=False)
+        self.assertEqual(os.environ.get('EBROOTGCC'), None)
+
```

Comment:
Well, the order does change, even ignoring hierarchies (at least with lmod, I am not sure with tmod):

```
~]$ ml

Currently Loaded Modules:
  1) GCCcore/.5.4.0 (H)   2) binutils/.2.28 (H)   3) StdEnv (H)   4) icc/.2017.2.174-GCC-5.4.0 (H)   5) ifort/.2017.2.174-GCC-5.4.0 (H)   6) Intel/2017.2.174-GCC-5.4.0

  Where:
   H:  Hidden Module



~]$ ml StdEnv
~]$ ml

Currently Loaded Modules:
  1) icc/.2017.2.174-GCC-5.4.0 (H)   2) ifort/.2017.2.174-GCC-5.4.0 (H)   3) Intel/2017.2.174-GCC-5.4.0   4) GCCcore/.5.4.0 (H)   5) binutils/.2.28 (H)   6) StdEnv (H)
```

I agree with you that we have no reason at all for getting rid of `$EBROOTXXX`. It is simply that instead of really checking if the module has been reloaded directly, you are checking for the byproducts of loading modules, which works fine, but is not the same. But if lmod and tmod behave differently here I think your check is good enough, no need to over-engineer it.